### ISSUE TITLE
Sprint September 18th - September 29th

### ISSUE THREAD

**Issue by mariusandra** [PostHog/posthog#17412]
# Global Sprint Planning

## 3 things that might take us down

1.
2.
3.

## Retro: What can we do better next sprint?

1.
2.
3.
4.
5.

# Team sprint planning

For your team sprint planning copy this template into a comment below for each team.

```
# Team ___

**Support hero:** ___

## Retro

<!-- Grab the high and low priority items from last time and add whether that item was completed or not -->

- 

## Hang over items from previous sprint

<!-- For each item, decide to re-prioritise (and add below) or deprioritise -->

- Item 1. prioritised/deprioritise

## OKR

1. OKR, status (red/yellow/green) and action points if yellow/red


### High priority

-

### Low priority / side quests

-

```

**Comment by mariusandra** [comment]
# Team ~Proud Syntactical~ Product Analytics

**Support hero:**
- Marius 1st week
- Michael 2nd week

## Retro

- Tom: great!
- Thomas: a lot of support. Deep dive into WAU/MAU issues.
- Marius: OK in general. Not as much of a deep dive into performance as I'd liked. Progress on HogQL conversion.

## Hang over items from previous sprint

- Porting over lifecycle and other insights into HogQL
- Performance and benchmarking.

## OKR

- 🟡 Port all our insights over to HogQL.
  - Got far with Lifecycle. Going from 0 to 1 was hard, 1 to 10 will be easier.
- 🔴 Sleeker insight editor UI
  - Not started

### High priority

- Convert all insights to HogQL

### Low priority / side quests

- Thomas: metadata migration & Zapier issue. 
- Marius: hook up performance benchmarks with HogQL

### Other notable things

https://ingesanagram.com/

**Comment by pauldambra** [comment]
# Team Manoel-itoring

**Support hero:** @daibhin 
David off half the sprint (@pauldambra cover support hero)

## Retro

* onboarding was fun and productive
* somehow ten degrees celsius warmer just that week
* felt productive despite a week onboarding
* Mr BLOBBY


## Hang over items from previous sprint

* ✅ blobby to 100% - @pauldambra and @benjackwhite 
* 🟢 insights experience in notebooks - @daibhin 
* ♻️ notebooks polish polish polish - all
    * how do we know we're finished - @annikaschmid enjoys working with it

### Low priority / side quests

* ~~carry-over notebooks sharing - @pauldambra~~
    * decided to leave this until we or someone else improve sharing permissions
* ✅ support feature success in notebooks usage - @daibhin 
* ✅ intelligent search across title and content - @pauldambra 

## Goals

🎯 Goal 1 : Notebooks on ProductHunt 🟢
🎯 Goal 2 : Next 5 great things for Replay 🟢
🥳 Filtering and searching UX => 🟡 done a lot here but more to do
🤔 "Deep dive" e.g. Funnels -> list -> recording => 🟡 we might get to this
🤔 Clips => 🟡 we might get to this
🤔 Mobile replay => 🔴 Manoel enables this in Q4

### High priority

* ♻️ ALL IN ON NOTEBOOKS - notebooks polish polish polish - @pauldambra @benjackwhite 
    * how do we know we're finished - @annikaschmid enjoys working with it
    * things in the #notebooks-feedback channel
    * can edit all the things
    * undo/redo/history @pauldambra 
    * playlists experience @benjackwhite 
* rewrite the Android SDKs to get the basis for mobile session replay @marandaneto 

### Low priority / side quests

* monitoring blobby
* documentation and rollout of longer replay TTL
* recordings proxy @benjackwhite 
* toolbar 3000 @pauldambra

**Comment by EDsCODE** [comment]
# Team Data Warehouse

### Retro

**Product**

- [ ] Work with anyone internally who is doing data analysis and get internal experience up to speed
     - [ ] common query paths (how to easily handle joins on json data)
     - [x] support JSON s3 type
- [x] MVP shipped

**Infra**

- [x] Wrap up HouseWatch deployments for EU / US 
- [x] Test backups made by HouseWatch
- [x] Setup incremental backup functionality on HouseWatch
- [ ] Make decision on CH.cloud vs hosted CH

### High priority

*Eric Off second week and first week of following sprint

**Product**
- [ ] release datawarehouse feature flag (not views!) and monitor usage
- [ ] convert revenue dashboard to posthog
- [ ] user outreach

**Infra**
- [ ] Finish doc & Make decision on CH.cloud vs hosted CH
- [ ] Test ClickPipes / Setup Vector
- [ ] Restore Persons table and benchmark different joins on CH.cloud
- [ ] Clean up ClickHouse EBS volumes as we taper off Session Recordings

**Comment by fuziontech** [comment]
# Team Data Warehouse

### Retro

**Product**
- [ ] Work with anyone internally who is doing data analysis and get internal experience up to speed
     - [ ] common query paths (how to easily handle joins on json data)
     - [ ] support JSON s3 type

Leftover:
- [ ] sanitize credentials from sentry
- [ ] Brainstorm methods to scale. Current iteration won't scale


**Infra**
- [x] Wrap up HouseWatch deployments for EU / US 
- [x] Test backups made by HouseWatch
- [x] Setup incremental backup functionality on HouseWatch
- [ ] Make decision on CH.cloud vs hosted CH

### High priority

**Product**


**Infra**
- [ ] Finish doc & Make decision on CH.cloud vs hosted CH
- [ ] Test ClickPipes / Setup Vector
- [ ] Restore Persons table and benchmark different joins on CH.cloud

**Comment by ellie** [comment]
# Team Infrastructure

**Support hero:** @frankh 

## Retro

- [ ] Get a mvp of capture-rs up and running @ellie
- [x] Fix current terraform drifts @danielxnj 
- [ ] Start splitting the state into smaller pieces organized by modules, to reduce the risk of corruption and speed up deployments @danielxnj 
- [x] switch fully to karpenter in US (hoping will reduce costs at least 10%) @frankh 
- [ ] finish up exit nodes and rotate VPN stuff (delete old VPN, get github actions using tailscale exit nodes, we have SSO VPN for backup access) @frankh 
- [ ] figure out our docker mirror suddenly costing $3k/month or replace it @frankh 

### ClickHouse work

- [x] Wrap up HouseWatch deployments for EU / US 
- [x] Test backups made by HouseWatch
- [x] Setup incremental backup functionality on HouseWatch
- [ ] Make decision on CH.cloud vs hosted CH

## This sprint
- capture is deployed to dev, still working on getting all the data conversion working @ellie
- get tiered kafka up and running for sessions @frankh 
- migrate our docker mirror @frankh 
- sort out new VPN + key rotation @frankh
- continue splitting the terraform state @danielxnj 
- upgrade EU aurora to 15 @danielxnj 

### ClickHouse Work

- [ ] Finish doc & Make decision on CH.cloud vs hosted CH
- [ ] Test ClickPipes / Setup Vector
- [ ] Restore Persons table and benchmark different joins on CH.cloud
- [ ] Clean up ClickHouse EBS volumes as we taper off Session Recordings

**Comment by raquelmsmith** [comment]
# Growth & Data Analysis

**Support hero:** Me

## Retro
TL;DR was very inundated with billing support this sprint so got little done otherwise.


- [ ] Build onboarding billing and "other products" steps @raquelmsmith 
  - Hope to have this done by friday but looking unlikely
- [ ] Change MRR per product reporting to be calculated based on usage, not invoices @raquelmsmith 
- [x] Understand where most performance issues are happening @stuti90 
- [ ] Start understanding what our real activation metric should be @stuti90 
- [x] https://github.com/PostHog/billing/pull/358
- [x] https://github.com/PostHog/billing/pull/360

## OKR

1. Starting to use and paying for new products feels frictionless
  - 🟡 In progress!
  - [ ] KR: We have separate onboarding for each product
  - [ ] KR: People don’t have to “upgrade” to different products, they just turn it on or start using it in some way.
3. Customers can pay a reasonable price for the value they’re receiving
  - 🟢 Done! Unless Data Warehouse or CDP become ready for launch.
  - [x] KR: Make sure rollout of new product analytics pricing goes well
  - [x] KR: Feature Flags pricing is launched, as well as Data Warehouse and CDP, if they are ready for launch

## This sprint
### High priority
- [x] Billing and "other products" onboarding steps @raquelmsmith
- [x] Feature flags onboarding @raquelmsmith 
- [x] Start understanding what our real activation metric should be @stuti90 

### Low priority / side quests

- [ ] Change MRR per product reporting to be calculated based on usage, not invoices @raquelmsmith

**Comment by neilkakkar** [comment]
# Team Feature Success

**Support hero:** @liyiy 

Neil: 1 day
Annika: 3 days

## Retro

1. Surveys: Calculate NPS, multiple questions, tests - @liyiy -> NPS & tests sorted, multiple questions will roll over
2. Continue building out components for notebook templates for flags here: https://github.com/PostHog/posthog/issues/17134 - @neilkakkar -> continuing, all 'create' components are done, edit + create standalone components remains. [Template building here](https://app.posthog.com/notebooks/vJJQxwuX).


## Hang over items from previous sprint

multiple questions
---

## OKRs

1. Make it easier for engineers to analyze the success of a feature
2. Build out an excellent user feedback product

### High priority

1. Ensure a successful onboarding for Juraj! - @everybody
2. Surveys: Multiple questions - @liyiy 
3. Surveys: whatever is needed to launch surveys pricing - @neilkakkar 

punting notebooks template to next sprint.

### Low priority / side quests

**Comment by tiina303** [comment]
# Team Pipeline

**Support hero:** @MarconLP effectively
**Vacation:** @xvello full sprint

## Retro



- @xvello - kuddo to Tiina for following up on many of the apps stuff, properties follow-ups ; why is PG memory usage growing - we're O(n) for a bunch of things and some O(nm), n teams, m properties per team; moving props, cohorts, ... out will help
- @bretthoerner - thanks Tiina/Xavier for pairing learning plugin-server context
- @tomasfarias - good sprint, kuddos to Marcus raising bugs from support
- @MarconLP - great sprint, workload a bit higher, but otherwise going pretty well
- @tiina303 - made good progress, seems like a fine sprint
- @timgl - +1 to Tiina great job with the deprecation stuff on the apps front

## Hang over items from previous sprint

- dedicated Redis in plugin-server - once @xvello  is back from vacation

## OKR

(no changes)

1. 🔴 Define and enforce limits on what users can send us - Owner @xvello 
2. 🟢 Deprecate and improve apps
    - 🟢 Finish batch exports so we can deprecate export apps - Owner @tomasfarias
    - 🟡 Remove all processing apps - Owner @tiina303
    	-> this has changed: we're keeping stateless apps, good progress and plan here for all the apps
3. 🟠  Figure out the pipeline architecture for the next 2 years - Owner @timgl
   - we have some stuff, but we don't have everything and export / webhook side is the focus atm

### High priority

- batch exports: autoscaling, alerting, dashboards, separate deployments @tomasfarias 
- unblock deleting exportEvents buffer from onEvent side (batch exports migration is the other blocker) @tiina303 
- RDKafka rollout @bretthoerner 

### Low priority / side quests

- batch exports: user facing improvements: logs, errors, metrics @tomasfarias 
- make local dev Kafka experience better @bretthoerner
